### PR TITLE
[C++] Remove namespace check for MultiTopicsConsumerImpl

### LINKED ISSUE #9449: Unable to subscribe to different topics on different namespaces with Python client
**Describe the bug**
Creating a consumer subscribed on different topics on different namespaces throws an error even if no regex pattern is used ( as per https://pulsar.incubator.apache.org/docs/en/concepts-messaging/#multi-topic-subscriptions )

**To Reproduce**
1. Create a custom tenant on pulsar with pulsar admin
```bin/pulsar-admin tenants create my-tenant```
2. Create 2 different namespaces on that tenant with pulsar admin
```
bin/pulsar-admin namespaces create my-tenant/ns1
bin/pulsar-admin namespaces create my-tenant/ns2
```
3. Create 2 topics, one per namespace with pulsar admin
```
bin/pulsar-admin topics create my-tenant/ns1/topic-1
bin/pulsar-admin topics create my-tenant/ns2/topic-2
```
4. Run this code
```python
import pulsar
from _pulsar import ConsumerType

def handle(consumer, message):
    print(message.value())

client = pulsar.Client(service_url="pulsar://localhost:36650")
consumer = client.subscribe(
            topic=["persistent://my-tenant/ns1/topic-1", "persistent://my-tenant/ns2/topic-2"],
            subscription_name="my-sub",
            consumer_type=ConsumerType.Shared,
            message_listener=handle,
            consumer_name="my-consumer-name"
        )
```

I get the error ERROR [0x11a3f7dc0] MultiTopicsConsumerImpl:737 | Different namespace name. expected: qa/users now:qa/accounts

I tried the same setup with the Java client and it works as expected

**Expected behavior**
The consumer should not throw an error and just connect correctly

**Desktop (please complete the following information):**
 - OS: MacOSX Catalina

**Additional context**
I tried with pulsar python client 2.6.1, 2.6.2 and 2.7.0 and all of them had the same issue

### PR DESCRIPTION
**Pull request by BewareMyPower**
Fixes #9449 

### Motivation

This is a catchup work for https://github.com/apache/pulsar/pull/5716 that supports multiple topic subscriptions across multiple namespaces.

### Modifications

- Move the check for namespace in `MultiTopicsConsumerImpl` to `PatternMultiTopicsConsumerImpl` that uses a regex subscription.
- Fix the existed tests for subscription on topics across different namespaces.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as *BasicEndToEndTest.testMultiTopicsConsumerDifferentNamespace*.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
And I've a question. Should we also add tests for Python client? Since it only affects `Client::subscribe` API and there're no related tests in Python tests before.

**Comment by BewareMyPower** [comment]
I found there's an existed Python test for multiple topics, I'll fix the test.

File: pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc [hunk 1]
```diff
@@ -33,7 +33,9 @@ PatternMultiTopicsConsumerImpl::PatternMultiTopicsConsumerImpl(ClientImplPtr cli
       patternString_(pattern),
       pattern_(std::regex(pattern)),
       autoDiscoveryTimer_(),
-      autoDiscoveryRunning_(false) {}
+      autoDiscoveryRunning_(false) {
+    namespaceName_ = TopicName::get(pattern)->getNamespaceName();
+}
```
Comment (zymap) [inline pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc:38]: Wrong format?

Comment (BewareMyPower) [reply pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc:38]: Did you mean code format?

Comment (zymap) [reply pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc:38]: Yes. The indent looks weird.

Comment (BewareMyPower) [reply pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc:38]: The C++ code format check automatically by CI, which uses `clang-format` and the `pulsar-client-cpp/.clang-format` to check the code style.

Comment (zymap) [reply pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc:38]: okay

**Review by zymap — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**